tensorpipe/CMakeLists.txt

```diff
@@ -4,133 +4,142 @@
-  channel/error.cc
-  channel/helpers.cc
-  common/address.cc
-  common/allocator.cc
-  common/error.cc
-  common/fd.cc
-  common/socket.cc
-  common/system.cc
-  core/context.cc
-  core/context_impl.cc
-  core/error.cc
-  core/listener.cc
-  core/listener_impl.cc
-  core/pipe.cc
-  core/pipe_impl.cc
-  transport/error.cc)
-
-set(TP_PUBLIC_HEADERS
-  tensorpipe.h
-  channel/context.h
-  channel/error.h
-  common/buffer.h
-  common/cpu_buffer.h
-  common/device.h
-  common/error.h
-  common/optional.h
-  core/context.h
-  core/error.h
-  core/listener.h
-  core/message.h
-  core/pipe.h
-  transport/context.h
-  transport/error.h)
-
-target_include_directories(tensorpipe PUBLIC
-  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
-  $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}>
-  $<INSTALL_INTERFACE:${TP_INSTALL_INCLUDEDIR}>)
-
+# TP_SRCS is the list of source files that we need to build libtensorpipe.
+set(TP_SRCS)
+set(TP_CUDA_SRCS)
+set(TP_ROCM_SRCS)
+
+# TP_HDRS is the list of header files that we need to install.
+set(TP_HDRS)
+set(TP_CUDA_HDRS)
+set(TP_ROCM_HDRS)
+
+list(APPEND TP_SRCS
+  "${CMAKE_CURRENT_SOURCE_DIR}/channel/error.cc"
+  "${CMAKE_CURRENT_SOURCE_DIR}/channel/helpers.cc"
+  "${CMAKE_CURRENT_SOURCE_DIR}/common/address.cc"
+  "${CMAKE_CURRENT_SOURCE_DIR}/common/allocator.cc"
+  "${CMAKE_CURRENT_SOURCE_DIR}/common/error.cc"
+  "${CMAKE_CURRENT_SOURCE_DIR}/common/fd.cc"
+  "${CMAKE_CURRENT_SOURCE_DIR}/common/socket.cc"
+  "${CMAKE_CURRENT_SOURCE_DIR}/common/system.cc"
+  "${CMAKE_CURRENT_SOURCE_DIR}/core/context.cc"
+  "${CMAKE_CURRENT_SOURCE_DIR}/core/context_impl.cc"
+  "${CMAKE_CURRENT_SOURCE_DIR}/core/error.cc"
+  "${CMAKE_CURRENT_SOURCE_DIR}/core/listener.cc"
+  "${CMAKE_CURRENT_SOURCE_DIR}/core/listener_impl.cc"
+  "${CMAKE_CURRENT_SOURCE_DIR}/core/pipe.cc"
+  "${CMAKE_CURRENT_SOURCE_DIR}/core/pipe_impl.cc"
+  "${CMAKE_CURRENT_SOURCE_DIR}/transport/error.cc"
+  )
+
+list(APPEND TP_HDRS
+  "${CMAKE_CURRENT_SOURCE_DIR}/tensorpipe.h"
+  "${CMAKE_CURRENT_SOURCE_DIR}/channel/context.h"
+  "${CMAKE_CURRENT_SOURCE_DIR}/channel/error.h"
+  "${CMAKE_CURRENT_SOURCE_DIR}/common/buffer.h"
+  "${CMAKE_CURRENT_SOURCE_DIR}/common/cpu_buffer.h"
+  "${CMAKE_CURRENT_SOURCE_DIR}/common/device.h"
+  "${CMAKE_CURRENT_SOURCE_DIR}/common/error.h"
+  "${CMAKE_CURRENT_SOURCE_DIR}/common/optional.h"
+  "${CMAKE_CURRENT_SOURCE_DIR}/core/context.h"
+  "${CMAKE_CURRENT_SOURCE_DIR}/core/error.h"
+  "${CMAKE_CURRENT_SOURCE_DIR}/core/listener.h"
+  "${CMAKE_CURRENT_SOURCE_DIR}/core/message.h"
+  "${CMAKE_CURRENT_SOURCE_DIR}/core/pipe.h"
+  "${CMAKE_CURRENT_SOURCE_DIR}/transport/context.h"
+  "${CMAKE_CURRENT_SOURCE_DIR}/transport/error.h"
+  )
 
 ## Channels
 
 ### basic
 
-target_sources(tensorpipe PRIVATE
-  channel/basic/channel_impl.cc
-  channel/basic/context_impl.cc
-  channel/basic/factory.cc)
-set(TP_PUBLIC_HEADERS ${TP_PUBLIC_HEADERS}
-  channel/basic/factory.h)
+list(APPEND TP_SRCS
+  "${CMAKE_CURRENT_SOURCE_DIR}/channel/basic/channel_impl.cc"
```

Comment:
Yes, I need to take care of this. I will update the changes 
- Keep another list and add through `target_sources()`
- Or keep the current `add_library()` call, add all other files through` target_sources()` since all are added a `PRIVATE`